### PR TITLE
figure titles

### PR DESCRIPTION
**Pull request by egolol**
It solves several subissues from issue #1205

### CODE REVIEW

**Review by annacanal — CHANGES_REQUESTED** [review]
After comparing a nodal measure, like Degree, I press to show comparison and an error jumps:

```
Brace indexing is not supported for variables of this type.

Error in PPComparisonGroup_Diff/update (line 954)
		                full_value{k, 2} = p1{k};

Error in PlotElement/update (line 1127)
		        pe.get('PR_DICT').getItem(prop).update()

Error in PlotElement/draw (line 1107)
		    pe.update()

Error in GUI/draw (line 1216)
		    pe.draw('Parent', gui.pp)

Error in PPCompareGroup_CPDict_WU/cb_measure_gui (line 968)
		                    pr.f_m{f_count} = GUI('pe', cp, 'POSITION', [x y w h], 'CLOSEREQ', false).draw();

Error in PPCompareGroup_CPDict_WU/draw/cb_measure_plot_btn (line 775)
		        pr.cb_measure_gui();
 
Error while evaluating UIControl Callback.
```

Moreover, titles of figures appear on atlas, but not on measure results figure 

atlas correct:
atlas_figure_titles_good

analysis no title:
measure_degree_plot_notitlename

**Comment by annacanal** [comment]
Rearrangement of diff, p1, p2 and ciu and cil works, now everything on the same table, but it is under the same title DIFF, and then we still have individually p2, p1...etc
Screenshot 2022-03-01 at 17 11 13

**Comment by annacanal** [comment]
In general, all the issues supposed to be fixed in this branch are fixed.
Except for:
1. adding **titles** on all **figures**, **plot brain view** from **analysis** or **comparison** is missing:
Screenshot 2022-03-01 at 23 33 23

 2. Mismatch for line width and marker width: **marker width does not work yet**, does not update size by its own in any of the plots from analysis or comparison of structural BUD


3. From the comparison, when pressing the buttons to add CIU and CIL on the plot across densities, these errors jump:

```
Output argument "limit" (and maybe others) not assigned during call to "PlotComparisonGroupLine.settings/obtain_cil_ciu_value".

Error in PlotComparisonGroupLine/settings/cb_show_confidence_interval_min (line 1338)
		                cil = obtain_cil_ciu_value(pr.get('CIL'));
 
Error while evaluating UIControl Callback.
```
```
Output argument "limit" (and maybe others) not assigned during call to "PlotComparisonGroupLine.settings/obtain_cil_ciu_value".

Error in PlotComparisonGroupLine/settings/cb_show_confidence_interval_max (line 1360)
		                ciu = obtain_cil_ciu_value(pr.get('CIU'));
 
Error while evaluating UIControl Callback.
```

**Review by annacanal — APPROVED** [review]

**Comment by annacanal** [comment]
All subissues meant to be solved in this branch have been tested. 
Some extra things detected are being taken care of by @egolol in a new branch (#1211 ). So this branch can be merged into gv_final

**Review by giovannivolpe — CHANGES_REQUESTED** [review]
* Substitute Inner_Show and Inner_Hide with cb_bring_to_front and cb_hide (change also in Plot)

File: braph2genesis/src/cohort/_PPSubject_ID.gen.m [hunk 1]
```diff
@@ -164,37 +164,7 @@ function cb_edit_value(pr)
 end
 function update_group(pr)
     figure_h = ancestor(pr.p, 'figure');
-    update_btn = get(figure_h, 'UserData');
```
Comment (giovannivolpe) [inline braph2genesis/src/cohort/_PPSubject_ID.gen.m:167]: do this with a callback function

**Review by giovannivolpe — CHANGES_REQUESTED** [review]

File: braph2genesis/pipelines/structural multiplex/_PPAnalyzeGroupMP_G.gen.m [hunk 2]
```diff
@@ -333,17 +333,26 @@ function cb_graph_value(pr)
 
     % TODO: check this part of the code once GUI is finalized
     value = el.getr(prop);
-    if isa(value, 'NoValue')
-        pr.f_g = GUI( ...
-            'PE', el.getPropDefault(prop), ...
-            'POSITION', [x y w h], ...
-            'CLOSEREQ', false).draw();
+    if isempty(pr.f_g) || ~check_graphics(pr.f_g, 'figure')
```
Comment (giovannivolpe) [inline braph2genesis/pipelines/structural multiplex/_PPAnalyzeGroupMP_G.gen.m:336]: use only check_graphics

File: braph2genesis/pipelines/structural multiplex/_PPAnalyzeGroupMP_G.gen.m [hunk 3]
```diff
@@ -392,21 +401,23 @@ function cb_measure_gui(pr)
...
-                    % hide
-                    set(tmp_f, 'Visible', 'off')
-                else
-                    % show
-                    figure(tmp_f)
-                    set(tmp_f, ...
-                        'Visible', 'on', ...
-                        'WindowState', 'normal' ...
-                        )
+            if isgraphics(tmp_f)
```
Comment (giovannivolpe) [inline braph2genesis/pipelines/structural multiplex/_PPAnalyzeGroupMP_G.gen.m:404]: use check_graphics

File: braph2genesis/src/analysis/_PPComparisonGroup_Diff.gen.m [hunk 4]
```diff
@@ -163,16 +167,27 @@ function update(pr)
...
 
+    %get brain atlas
+    br_dict = el.get('C').get('A1').get('g').get('brainatlas').get('br_dict');
+    br_ids = cellfun(@(x) x.get('id'), br_dict.getItems(), 'UniformOutput', false);
+
+
     if  size(value, 2) > 2
         set(pr.slider_text, ...
-            'String', [label ' ' num2str(round(get(pr.slider, 'Value')))]);
+            'String', [label ': ' slider_tags{round(get(pr.slider, 'Value'))* map_multiplier}]);
```
Comment (giovannivolpe) [inline braph2genesis/src/analysis/_PPComparisonGroup_Diff.gen.m:177]: space

File: braph2genesis/src/analysis/_PPComparisonGroup_Diff.gen.m [hunk 5]
```diff
@@ -163,16 +167,27 @@ function update(pr)
...
         set(pr.second_slider_text, ...
-            'String', [label ' ' num2str(round(get(pr.second_slider, 'Value')))]);
+            'String', ['Layer: ' num2str(round(get(pr.second_slider, 'Value'))* map_multiplier)]);
```
Comment (giovannivolpe) [inline braph2genesis/src/analysis/_PPComparisonGroup_Diff.gen.m:179]: space

File: braph2genesis/src/analysis/_PPComparisonGroup_Diff.gen.m [hunk 6]
```diff
@@ -56,17 +56,18 @@
...
         'Units', 'characters', ...
         'Visible', 'off', ...
-        'Value', max(1, L), ...
-        'Min', 1, ...
-        'Max', L, ...
-        'SliderStep', [1 1], ...
+        'Value', 1 / map_multiplier, ...
+        'Min', 1 / map_multiplier, ...
+        'Max', L / map_multiplier, ...
+        'SliderStep', [map_multiplier/(L*map_multiplier) map_multiplier/(L*map_multiplier)] , ...
```
Comment (giovannivolpe) [inline braph2genesis/src/analysis/_PPComparisonGroup_Diff.gen.m:70]: spaces

File: braph2genesis/src/analysis/_PPComparisonGroup_Diff.gen.m [hunk 7]
```diff
@@ -91,11 +92,11 @@ function cb_slider(~, ~)
...
             'Units', 'characters', ...
-            'Value', 1, ...
-            'Min', 1, ...
-            'Max', L2, ...
-            'SliderStep', [1 1], ...
-            'Callback', {@cb_slider_2} ...
+            'Value', 1 / map_multiplier, ...
+            'Min', 1 / map_multiplier, ...
+            'Max', L2 / map_multiplier, ...
+            'SliderStep', [map_multiplier/(L2*map_multiplier) map_multiplier/(L2*map_multiplier)] , ...
```
Comment (giovannivolpe) [inline braph2genesis/src/analysis/_PPComparisonGroup_Diff.gen.m:98]: spaces

File: braph2genesis/src/atlas/_PlotBrainAtlas.gen.m [hunk 8]
```diff
@@ -1839,7 +1839,8 @@ function cb_interpreter(~,~)  %  (src,event)
         f_out = f;
     end
 end
-function Inner_Show(pl)
+function cb_bring_to_front(pl)
+    pl.cb_bring_to_front@Plot();
```
Comment (giovannivolpe) [inline braph2genesis/src/atlas/_PlotBrainAtlas.gen.m:1843]: comment

**Review by giovannivolpe — CHANGES_REQUESTED** [review]

**Review by giovannivolpe — APPROVED** [review]

**Pull request merged**